lib/template.html

```diff
@@ -131,18 +151,27 @@
   <meta property="og:site_name" content="{{ seo_site_title }}" />
 {% endif %}
 
-{% if seo_page_image %}
-<meta property="og:image" content="{{ seo_page_image }}" />
-  {% if page.image.height %}
-    <meta property="og:image:height" content="{{ page.image.height }}" />
+{% if page.image %}
+  {% if page.image.facebook %}
+    {% assign local_seo_page_image = seo_page_image_facebook %}
+    {% assign local_seo_page_image_height = seo_page_image_facebook_height %}
+    {% assign local_seo_page_image_width = seo_page_image_facebook_width %}
+  {% else %}
+    {% assign local_seo_page_image = seo_page_image_default %}
```

Comment:
`local_` is used in this case to assign one of the images (and create a fallback in case there is no facebook image) inside of `{% if page.image %}`. It's only used in lines 164 - 169. I pull out the image variable here because if we only checked (for example) the facebook image, but didn't add dimension (and did add dimensions to the `default` image) we would have a facebook image assigned dimension tags of the default image. So here I make sure that each image type is properly assigned its image values from the front matter, and no more. 

Having it before the *meta tag logic* makes sure that one can read the logic below without losing themselves in multiple *if statements* each time. 

Maybe using something other than the name `local_` might help, but I can't think of something that wouldn't load the variable name too much..